### PR TITLE
feat: adjust styling for search filters

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Flex should flow to next lines when not enough space:
Screenshot 2023-12-11 at 12 39 05

When more than 2 items are selected we do +n text:
Screenshot 2023-12-11 at 12 37 31

On very small screens or with very long item names we can fall back to ellipsis:
Screenshot 2023-12-11 at 13 04 28

### CODE REVIEW

File: frontend/src/component/common/FilterItem/FilterItem.test.tsx [hunk 1]
```diff
@@ -67,6 +67,19 @@ describe('FilterItem Component', () => {
...
 
+    it('renders explicit and extra options', async () => {
+        const mockState = {
+            operator: 'IS_ANY_OF',
+            values: ['1', '3', '2'],
+        };
+
+        const recordedChanges = setup(mockState);
+
+        const valuesElement = await screen.findByText('1, 3 +1');
```
Comment (kwasniew) [inline frontend/src/component/common/FilterItem/FilterItem.test.tsx:78]: testing is the last item will generate +1

File: frontend/src/component/common/FilterItem/FilterItemChip/FilterItemChip.tsx [hunk 2]
```diff
@@ -40,6 +40,13 @@ const StyledCategoryIconWrapper = styled('div')(({ theme }) => ({
 const StyledOptions = styled('span')(({ theme }) => ({
     color: theme.palette.text.primary,
     fontWeight: theme.typography.fontWeightRegular,
+    textOverflow: 'ellipsis',
```
Comment (kwasniew) [inline frontend/src/component/common/FilterItem/FilterItemChip/FilterItemChip.tsx:43]: extra protection is 2 explicit items and +n would be too long we need ellipsis. Ellipsis always requires the whiteSpace no wrap, overflow hidden and explicit width.

File: frontend/src/component/feature/FeatureToggleList/FeatureToggleFilters/FeatureToggleFilters.tsx [hunk 3]
```diff
@@ -11,6 +11,7 @@ const StyledBox = styled(Box)(({ theme }) => ({
     display: 'flex',
     padding: theme.spacing(2, 3),
     gap: theme.spacing(1),
+    flexWrap: 'wrap'
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureToggleList/FeatureToggleFilters/FeatureToggleFilters.tsx:14]: this moves new filters to new lines when not enough space

File: frontend/src/component/feature/FeatureToggleList/FeatureToggleFilters/FeatureToggleFilters.tsx [hunk 4]
```diff
@@ -100,8 +101,8 @@ export const FeatureToggleFilters: VFC<IFeatureToggleFiltersProps> = ({
                 filterKey: 'segment',
                 singularOperators: ['INCLUDE', 'DO_NOT_INCLUDE'],
                 pluralOperators: [
-                    'INCLUDE_ALL_OF',
                     'INCLUDE_ANY_OF',
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureToggleList/FeatureToggleFilters/FeatureToggleFilters.tsx:104]: any of is more logical as a default choice

**Review by sjaanus — APPROVED** [review]

**Pull request merged**